tests/SerializationTest.php

```diff
@@ -164,4 +168,23 @@ public function testDatesInBodyParametersAreSerializedInZuluFormat(): void
 
         $this->assertEquals('2024-01-01T00:00:00Z', $body['packageDetail']['shipDate']);
     }
+
+    public function testDeserializeDateTimeWithTimezone(): void
+    {
+        $now = new DateTime();
+        $now->setTimeZone(new DateTimeZone('+02:00'));
+
+        $result = GetMyFeesEstimateResponse::deserialize([
+            'payload' => [
+                'FeesEstimateResult' => [
+                    'status' => 'Success',
+                    'FeesEstimate' => [
+                        'TimeOfFeesEstimation' => $now->format(DateTimeInterface::ATOM),
+                    ],
+                ],
+            ],
+        ]);
+        $this->assertNotNull($result);
+        $this->assertInstanceOf(DateTimeInterface::class, $result->payload->feesEstimateResult->feesEstimate->timeOfFeesEstimation);
```

Comment:
Part of what we're trying to test here is that the deserialized DateTime actually has the _right_ timezone. Could you add a test that checks that that's happening here?

Reply:
Added additionally the check for the right timezone.